sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxReportSensor.java

```diff
@@ -152,24 +149,30 @@ protected String getStringProperty(String name, String def) {
 
     List<File> reports = new ArrayList<>();
 
-    List<String> includes = Arrays.asList(settings.getStringArray(reportPathPropertyKey));
-    if (!includes.isEmpty()) {
-      includes = Lists.transform(includes, new Function<String, String>() {
-        @Override
-        public String apply(@Nullable String path) {
-          if (path == null){
-            return null;
-          }
-          path = FilenameUtils.normalize(path);
-          if (new File(path).isAbsolute()) {
-            return path;
-          }
-          return FilenameUtils.normalize(moduleBaseDir.getAbsolutePath() + File.separator + path);
+    List<String> reportPaths = Arrays.asList(settings.getStringArray(reportPathPropertyKey));
+    if (!reportPaths.isEmpty()) {
+      List<String> includes = new ArrayList<>();
+      for (String reportPath : reportPaths) {
+        // Normalization can return null if path is null, is invalid, or is a path with back-ticks outside known directory structure
+        String normalizedPath = FilenameUtils.normalize(reportPath);
+        if (normalizedPath != null && new File(normalizedPath).isAbsolute()) {
+          includes.add(normalizedPath);
+          continue;
```

Comment:
@mjdetullio I'm not a fan of continue in loops. Think if / else if / else would be the same ....